test/univariate_statistics_test.cpp

```diff
@@ -833,6 +834,59 @@ void test_interquartile_range()
     BOOST_TEST_EQ(iqr, 6);
 }
 
+template<class Z>
+void test_mode()
+{
+    std::vector<Z> v {1, 2, 2, 3, 4, 5};
+    const Z ref = 2;
+
+    // Does iterator call work?
+    std::vector<Z> test = boost::math::statistics::mode(v.begin(), v.end());
+    BOOST_TEST(ref == test[0]);
+
+    // Does container call work?
+    test = boost::math::statistics::mode(v);
+    BOOST_TEST(ref == test[0]);
+
+    // Does it work with part of a vector?
+    test = boost::math::statistics::mode(v.begin(), v.begin() + 3);
+    BOOST_TEST(ref == test[0]);
+
+    // Does it work with const qualification? Only if pre-sorted
+    test = boost::math::statistics::sorted_mode(v.cbegin(), v.cend());
+    BOOST_TEST(ref == test[0]);
+
+    // Does it work with std::array?
+    std::array<Z, 6> u {1, 2, 2, 3, 4, 5};
+    test = boost::math::statistics::mode(u);
+    BOOST_TEST(ref == test[0]);
```

Comment:
I would use `BOOST_TEST_EQ` here rather than `BOOST_TEST`; it'll give a nicer message if we ever break this.